src/components/AdminForm.js

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import { connect } from "react-redux";
+import { setCohortName, setCohortType } from "../redux/actions/adminFormActions";
+
+import TextInput from "./TextInput";
+import Select from "./Select";
+import SubmitButton from "./SubmitButton";
+
+const AdminForm = (props) => {
+
+  const handleCohortNameChange = e => {
+    props.setCohortName(e.target.value);
+  };
+
+  const handleCohortTypeChange = (e) => {
+    props.setCohortType(e.target.value);
+  }
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    alert("Submitting");
+    // on submit we need to save the form in DB
+  }
+
+  const selectOptions = [
+    { value: "frontend", displayedName: "Fronend" },
+    { value: "backend", displayedName: "Backend" },
+    { value: "productDesign", displayedName: "Product Design" }
+  ];
+
+  return (
+    <div>
+      <form onSubmit={handleSubmit}>
+        <TextInput value={props.cohortName} handleChange={handleCohortNameChange} />
+        <Select
+          value={props.cohortType}
+          handleChange={handleCohortTypeChange}
+          options={selectOptions}
+        />
+        <SubmitButton />
+      </form>
+    </div>
+  );
+}
+
+const mapStateToProps = state => {
+  return {
+    cohortName: state.cohortInfo.adminFormReducer,
+    cohortType: state.cohortInfo.adminFormReducer
+  };
+};
+
+const mapDispatchToProps = (dispatch, getState) => {
```

Comment:
you dont need to pass getState here and we only dispatching on actions

Reply:
also you dont need dispatch here, just write the name of the function..try this: this one works as well:

Reply:
const mapDispatchToProps = {
  setCohortName,
  setCohortType
};

Reply:
usually we write in the way that you wrote when we use thunk.. thats based on what I learned.. but you can ask your mentor as well :-)

Reply:
I've removed the getState as it was not used, you're right. I am however keeping the dispatch because I am not using thunks for these actions as they are direct. Without it being a thunk, the syntax above doesn't work so I kept it.